crates/providers/lattice-controller/src/lib.rs

```diff
@@ -0,0 +1,630 @@
+            .get_client(&arg.lattice_id)
+            .await?
+            .perform_provider_auction(&arg.provider_ref, &arg.link_name, arg.constraints.clone())
+            .await
+            .map(|v| {
+                v.into_iter()
+                    .map(|a| ProviderAuctionAck {
+                        host_id: a.host_id,
+                        link_name: a.link_name,
+                        provider_ref: a.provider_ref,
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?)
+    }
+
+    /// Auction an actor on the lattice
+    #[instrument(level = "debug", skip(self, _ctx), fields(actor_id = ?_ctx.actor, lattice_id = ?arg.lattice_id))]
+    async fn auction_actor(
+        &self,
+        _ctx: Context,
+        arg: ActorAuctionRequest,
+    ) -> ProviderInvocationResult<Vec<ActorAuctionAck>> {
+        Ok(self
+            .get_connections()
+            .await
+            .get_client(&arg.lattice_id)
+            .await?
+            .perform_actor_auction(&arg.actor_ref, arg.constraints.clone())
+            .await
+            .map(|v| {
+                v.into_iter()
+                    .map(|a| ActorAuctionAck {
+                        actor_ref: a.actor_ref,
+                        host_id: a.host_id,
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?)
+    }
+
+    /// Retrieve all hosts on the lattice
+    #[instrument(level = "debug", skip(self, _ctx, arg), fields(actor_id = ?_ctx.actor, lattice_id = %arg.to_string()))]
+    async fn get_hosts(&self, _ctx: Context, arg: String) -> ProviderInvocationResult<Vec<Host>> {
+        Ok(self
+            .get_connections()
+            .await
+            .get_client(&arg.to_string())
+            .await?
+            .get_hosts()
+            .await
+            .map(|v| {
+                v.into_iter()
+                    .map(|h| Host {
+                        cluster_issuers: h.cluster_issuers,
+                        ctl_host: h.ctl_host,
+                        id: h.id,
+                        js_domain: h.js_domain,
+                        labels: h.labels,
+                        lattice_prefix: h.lattice_prefix,
+                        prov_rpc_host: h.rpc_host.clone(),
+                        rpc_host: h.rpc_host,
+                        uptime_human: h.uptime_human,
+                        uptime_seconds: h.uptime_seconds,
+                        version: h.version,
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?)
+    }
+
+    /// Retrieve inventory for a given host on the lattice
+    #[instrument(level = "debug", skip(self, _ctx, arg), fields(actor_id = ?_ctx.actor, host = ?arg.host_id, lattice_id = ?arg.lattice_id))]
```

Comment:
Nit: change to `skip_all` (last comment of this, same nit everywhere 😅)

Reply:
- [ ] Add `skip_all`